lightnode/docker/Dockerfile

```diff
@@ -0,0 +1,41 @@
+FROM golang:1.21.12-bookworm AS base
+
+# Install core dependencies
+RUN apt update
+RUN apt install -y build-essential
+
+# Set up lnode user
+RUN useradd -m -s /bin/bash lnode
+USER lnode
+WORKDIR /home/lnode
+# Remove default crud
+RUN rm .bashrc
+RUN rm .bash_logout
+RUN rm .profile
+
+# Copy the eigenda repository
+COPY --chown=lnode . /home/lnode/eigenda
+
+# Download all go dependencies and build the binary.
+WORKDIR /home/lnode/eigenda/lightnode
+RUN go mod download
```

Comment:
I think a worthwhile optimization would be the following sequence:
1. copy go.mod
2. Run `go mod download`
3. Copy the remainder of source
4. Run build

This would prevent having to redownload the go modules every time the source changes (as long as modules don't change)

Reply:
Very good suggestion. Change made.